### ISSUE TITLE
Files uploaded in front-end entry form are lost

### ISSUE THREAD

**Issue by mrw** [aelvan/Preparse-Field-Craft#57]
Hi! Thanks for your great work on Preparse. I think this issue is related to #23 and #52. 

Starting in 1.0.7, files uploaded as part of a front-end entry form aren't working for us, with Craft 3.1->3.3. The entry is getting saved properly but the file is disappearing. I tracked this down to [the `resetUploads()` method in PreparseField.php](https://github.com/aelvan/Preparse-Field-Craft/blob/70fdbc74f56ea0a64432f5251859a892fa1656d0/src/PreparseField.php#L162-L166) that was added.

I know you changed it from `unset($_FILES);` to `$_FILES = [];` to fix some issues, but that's still a problem for us because the file is getting lost. I hacked a fix for it for now (by adding a hidden field to our entry form and then returning early from `resetUploads()` if it's present) but I know that's not a permanent fix if we upgrade in the future. 

Do you have any ideas on other fixes for this?

Thanks!

**Comment by sjcallender** [comment]
@mrw Mind sharing your temp fix for this?

**Comment by mrw** [comment]
@sjcallender yes, in `preparse-field/src/PreparseField.php` we added this to the top of `resetUploads()` (you can fork it then reference your own fork in composer):

```php
if (!empty($_POST['is_fe_upload_form']) && $_POST['is_fe_upload_form'] == 'yes') {
    return;
}
```

Then in our front-end form we added this:

```html
<input type="hidden" name="is_fe_upload_form" value="yes">
```

It's not great but it works!

**Comment by sjcallender** [comment]
Thanks @mrw!

**Comment by adrienne** [comment]
Has this ever gotten fixed, btw?

**Comment by adrienne** [comment]
Hey @aelvan, has this behavior ever been resolved?

**Comment by adrienne** [comment]
Er, sorry, @aelvan, i have no idea why i tagged you. Should've been @jalendport!

**Comment by ademers** [comment]
I just discovered the same issue. However, only with uploading new user photos in front-end user profile forms as per https://craftcms.com/knowledge-base/front-end-user-accounts#user-profile-form. In `preparse-field/src/PreparseField.php`, I did the following based on comment above https://github.com/besteadfast/craft-preparse-field/issues/57#issue-490492389, and that seems to have resolved it. Preparse 1.2.5 on Craft 3.7.18.2
```
  private function resetUploads()
  {
      // $_FILES = [];
      unset($_FILES);
      // UploadedFile::reset();
  }
```

**Comment by engram-design** [comment]
Also seeing this behaviour, exactly as @ademers describes it.

**Comment by leevigraham** [comment]
This still seems to be an issue in CraftCMS 4.6.0 and Preparse 2.0.2.
